src/lib/db/store.ts

```diff
@@ -229,7 +231,20 @@ export function createDbStoreCreator(
                 ])
               }
 
-        // TODO: Reflect tags
+        const tagMap = storage.tagMap
+        await Promise.all(
+          noteDoc.tags.map(async tag => {
+            if (tagMap[tag] == null) {
+              tagMap[tag] = {
```

Comment:
We should not modify store state directly.

Reply:
changed, tell me if that's a bit too lenghty